### PR TITLE
Query editor input titles

### LINKED ISSUE #9115: Untitled documents have different naming convention based on how created
- Azure Data Studio Version: `master`
 
Steps to Reproduce:

1. Create an untitled SQL editor using New Query command on node in OE
2. Create an untitled SQL editor using New Query command from File menu

Expected: Both documents follow same naming convention

Actual: One is named `SQLQuery_1` the other is named `Untitled-1`

Screen Shot 2020-02-11 at 10 28 46 AM

### PR DESCRIPTION
**Pull request by anthonydresser**
This PR addresses #9115

### CODE REVIEW

File: src/sql/workbench/browser/parts/editor/editorStatusModeSelect.ts [hunk 1]
```diff
@@ -41,7 +42,7 @@ export async function setMode(accessor: ServicesAccessor, modeSupport: IModeSupp
 		if (newInputCreator) { // if we know how to handle the new language, tranform the input and replace the editor (e.x notebook, sql, etc)
 			const newInput = newInputCreator.convertInput(input || activeEditor);
 			if (newInput) {  // the factory will return undefined if it doesn't know how to handle the input
-				await editorService.replaceEditors([{ editor: activeEditor, replacement: newInput }], activeControl.group);
+				await editorService.replaceEditors([{ editor: activeEditor, replacement: isThenable(newInput) ? await newInput : newInput }], activeControl.group);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/browser/parts/editor/editorStatusModeSelect.ts]: We could just add await here, right? It just wouldn't do anything for the non-promise input. (I don't really have a preference - just confirming that my understanding is correct here)

File: src/sql/workbench/services/languageAssociation/common/doHandleUpgrade.ts [hunk 2]
```diff
@@ -20,8 +20,8 @@ export function doHandleUpgrade(editor?: EditorInput): EditorInput | undefined {
 			editor.getPreferredMode();
 		}
 		const association = languageRegistry.getAssociationForLanguage(language);
-		if (association) {
-			return association.convertInput(editor);
+		if (association && association.syncConvertinput) {
```
Comment (Charles-Gagnon) [inline src/sql/workbench/services/languageAssociation/common/doHandleUpgrade.ts]: There's other places we call convertInput - you aren't going to update those?

File: src/sql/workbench/browser/parts/editor/editorStatusModeSelect.ts (hunk 1 above)
Comment (anthonydresser) [reply src/sql/workbench/browser/parts/editor/editorStatusModeSelect.ts]: you can't await something that isn't a promise.

File: src/sql/workbench/services/languageAssociation/common/doHandleUpgrade.ts (hunk 2 above)
Comment (anthonydresser) [reply src/sql/workbench/services/languageAssociation/common/doHandleUpgrade.ts]: this is the only one that can't handle a promise. the other places we call convert input I've updated to handle promise.

File: src/sql/workbench/contrib/query/browser/queryInputFactory.ts [hunk 3]
```diff
@@ -31,9 +32,37 @@ export class QueryEditorLanguageAssociation implements ILanguageAssociation {
...
+		@IQueryEditorService private readonly queryEditorService: IQueryEditorService) { }
 
-	convertInput(activeEditor: IEditorInput): QueryEditorInput | undefined {
+	async convertInput(activeEditor: IEditorInput): Promise<QueryEditorInput | undefined> {
+		const queryResultsInput = this.instantiationService.createInstance(QueryResultsInput, activeEditor.resource.toString(true));
+		let queryEditorInput: QueryEditorInput;
+		if (activeEditor instanceof FileEditorInput) {
+			queryEditorInput = this.instantiationService.createInstance(FileQueryEditorInput, '', activeEditor, queryResultsInput);
+		} else if (activeEditor instanceof UntitledTextEditorInput) {
+			const content = (await activeEditor.resolve()).textEditorModel.getValue();
```
Comment (anthonydresser) [inline src/sql/workbench/contrib/query/browser/queryInputFactory.ts]: This is basically the same function as the sync version, except promised so we can do this.

File: src/sql/workbench/browser/parts/editor/editorStatusModeSelect.ts (hunk 1 above)
Comment (Charles-Gagnon) [reply src/sql/workbench/browser/parts/editor/editorStatusModeSelect.ts]: Is the strict compile blocking that or something? It's perfectly fine to await a non-promise value in normal Typescript. It just doesn't do anything useful. But it's not like we're really gaining anything by using isThenable here.

https://github.com/Microsoft/TypeScript/issues/8310

Comment (anthonydresser) [reply src/sql/workbench/browser/parts/editor/editorStatusModeSelect.ts]: maybe i was interpreting the error i was getting wrong, ill take a look

**Pull request merged**